tests/unit/js/test.ol.js

```diff
@@ -10,19 +10,27 @@ beforeEach(() => {
     sandbox.stub(global, '$').callsFake(jquery);
 });
 
-test('initReadingListFeature binds events to HTML added after initialisation', () => {
-    const spy = sinon.spy();
-    initReadingListFeature(() => spy);
-    $(document.body).html(`<div>
-    <div class="dropclick">test</div>
-    <a class="add-to-list"></a>
-</div>
-    `);
-    // trigger a click event
-    $('.dropclick').trigger('click');
-    // check the event handler was called
-    expect(spy.callCount).toBe(1);
-    $('.dropclick').trigger('click');
-    // check the dropclick event handler was called
-    expect(spy.callCount).toBe(2);
+describe('initReadingList', () => {
+    test('dropdown changes arrow direction on click', () => {
+        const clock = sinon.useFakeTimers();
+        initReadingListFeature();
+        $(document.body).html(`
+            <a href="javascript:;" class="dropclick dropclick-unactivated">
+                <div class="arrow arrow-unactivated"></div>
+            </a>
+        `);
+
+        const $dropclick = $('.dropclick');
+        const $arrow = $dropclick.find('.arrow');
+
+        for (let i = 0; i < 2; i++) {
+            $dropclick.trigger('click');
+            clock.next(); // need to step forward because using debounce
+            expect($arrow.hasClass('up')).toBe(true);
+            $dropclick.trigger('click');
+            clock.next();
+            expect($arrow.hasClass('up')).toBe(false);
```

Comment:
This is much better than before as it tests the UI changes in response to this action.
I'm not sure how complete you want your changes to be, but I'd love to see the click handlers pulled out into separate testable functions

e.g.
```
$(document).on('change', '.reading-log-lite select', function(e) {
```
becomes
```
$(document).on('change', '.reading-log-lite select', makeSelectChangeHandler( ... ) );
```

No need to do that now, just want to check in we're working towards a  similar goal.

Reply:
I haven't though about it too much, but probably something like this:

```js
class ReadingListDropdown {
  constructor() {
    // bind events
    $(document).on('change', '.blah foo', this.handleTheThingThatChange___YouKnowTheThing);
  }
  handleTheThingThatChange___YouKnowTheThing(...) { ... }
}
```

But yeah, probably a separate PR :P

Reply:
WON'T FIX